### PR TITLE
Prefix support

### PR DESCRIPTION
**Pull request by sjadhav25**

### CODE REVIEW

**Comment by drsherluck** [comment]
Did you test this with a plan? In for example devops-toolkit repo?

**Comment by sjadhav25** [comment]
Yes i tested it in devops-toolkit repo

**Comment by sjadhav25** [comment]
Screenshot of plan where this module is used, 

image

File: instance.tf [hunk 1]
```diff
@@ -1,8 +1,8 @@
 resource "google_sql_database_instance" "instance" {
   database_version     = var.database_version
-  name                 = var.instance_name
+  name                 = "${var.project_prefix}-${var.instance_name}"
   deletion_protection  = var.deletion_protection
-  master_instance_name = var.primary_instance_name
+  master_instance_name = "${var.primary_instance_name}" != null ? "${var.project_prefix}-${var.primary_instance_name}" : null
```
Comment (tun0) [inline instance.tf]: I kinda doubt whether we should add the prefix here as well. Only other use case would be "cross project" replication 🤔

Comment (sjadhav25) [reply instance.tf]: Is it not good to have it if we use replication in future or it is the case we will never use cross project replication. Then i can remove prefix thing for primary instance.

Comment (tun0) [reply instance.tf]: I was mostly just "thinking out loud".

Guess we should leave it like this, for now. Could always introduce something like `external_primary_instance_name` with more flexibility down the road, if we'd ever need that.

**Review by tun0 — APPROVED** [review]

**Comment by drsherluck** [comment]
I think we should wait with the merging. we probably want to have to versions one for the current instances and one for the new ones that going to be created later. `instance_name` changes forces a recreate if I am not mistaken. So we should only use these modules when we are migrating the instances to the new project.

imo

**Comment by sjadhav25** [comment]
Added version v1.0.0 to main and published the module(current module without prefix support).

File: instance.tf (hunk 1 above)
Comment (tun0) [inline instance.tf]: If `project_prefix` is `null`, you don't want the `-` separator. As that would make the name start with a dash, which is both undesired and likely not supported either.

The cleanest solution might be to move the conditional logic to some locals and then refer to those locals here instead. If that'd be desired kinda depends on the amount of "magic" required to get the proper values.

Comment (tun0) [reply instance.tf]: Current "fix" is not what I meant. What I meant was:

If `project_prefix` is defined, set `name` to `instance_name` prefixed with `project_prefix`, separated with a dash.
If `project_prefix` is not defined, just set `name` to `instance_name` (and thus no dash).

Similar logic for `master_instance_name`, which will then need nested conditional logic (as the value depends on both the presence of `primary_instance_name` and `project_prefix`).

**Pull request merged**